### PR TITLE
Update + add functional option WithContext

### PR DESCRIPTION
**Pull request by pandatix**
**What does this PR contains ?**
 - an update of the `go.mod`: after removing unit/integration tests, the dependencies used were not removed, making the Go module heavier for no reason.
 - functional option `WithContext`: during implementation, you may need to manipulate the request's context (for timeout, cancel function...). Such option empowers the developer to do so. Moreover, as options are... optional, you can not specify any and still run your code. For reference, please see https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis and https://github.com/uber-go/guide/blob/master/style.md#functional-options.

Notice this PR is **backward compatible**.

### CODE REVIEW

**Comment by juliensalinas** [comment]
Hey @pandatix, thanks for this!
Let me have a look ASAP!

File: api.go [hunk 1]
```diff
@@ -10,9 +10,9 @@ type AdGenerationParams struct {
 }
 
 // AdGeneration generates a product description or an ad by contacting the API.
-func (c *Client) AdGeneration(params AdGenerationParams) (*AdGeneration, error) {
+func (c *Client) AdGeneration(params AdGenerationParams, opts ...Option) (*AdGeneration, error) {
```
Comment (juliensalinas) [inline api.go:13]: Hey @pandatix !
This PR looks great to me. I only have 1 question: in your opinion, is it best to set up these options at the client creation level (i.e. in `NewClient()`) or at the request level like you did?
I don't have an opinion on this. I'm simply asking to know if you thought about the 2 alternatives.
Thanks in advance!

Comment (pandatix) [reply api.go:13]: Hi Julien, as it's pointed out by Dave Cheney (https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) and multiple others on this subject, it's better on the function itself rather than the whole refering object, in this case the client.
Indeed, the part we use a context for is the query's one, and even if the client is executing the request, it's not what needs to be optioned. Nevertheless, some use cases may need to have options on the client.

Let's give an example: you have a job that runs multiple pipelines which uses the NLP-Cloud API for text analysis, translations, keywords extraction or whatever your business needs to perform efficiently. Your client will be initialized at the root of your job, shared between your pipelines and calls. Then you call the functions you need, but you don't want a call to last too long (e.g. 30 seconds) or want to be able to stop the calls properly (os.Signal, control condition...). This won't be performed at the root (the client, only holding the underlying HTTP client and access token) as it does not have a deadline or a control condition, but at the function (which performs the call using the client). This is an example for the context, but depending on specific use-cases of future customers, other options may be interesting to have.
For more info (and actual code samples), check Dave's blog post and Uber go-guide documentation (or even the Go grpc module...etc.).

TL;DR: what needs to be context-refered ? The function that the developer calls 😉

Comment (juliensalinas) [reply api.go:13]: Ok you 100% convinced me!
Actually that's the kind of behavior I can see in most of the libs I'm using that are making external API calls.
On NLP Cloud it would make perfect sense to do that at the request level because some requests are returning very slowly by nature (blog post generation for example) while others are returning very fast (embeddings for example). So the user might definitely want to set up different contexts in such cases.
Thanks again Lucas, your help is very much appreciated!
